lib/ledger_sync/adaptors/ledger_serializer_attribute.rb

```diff
@@ -0,0 +1,150 @@
+# frozen_string_literal: true
+
+require_relative 'ledger_serializer_type/value'
+Gem.find_files('ledger_sync/adaptors/ledger_serializer_type/**/*.rb').each { |path| require path }
+
+module LedgerSync
+  module Adaptors
+    class LedgerSerializerAttribute
+      attr_reader :block,
+                  :id,
+                  :ledger_attribute,
+                  :resource_attribute,
+                  :resource_class,
+                  :serializer,
+                  :type
+
+      def initialize(block: nil, id: false, ledger_attribute:, resource_attribute: nil, resource_class: nil, serializer: nil, type: LedgerSerializerType::Value)
+        raise 'block and resource_attribute cannot both be present' unless block.nil? || resource_attribute.nil?
+
+        @block = block
+        @id = id
+        @ledger_attribute = ledger_attribute.to_s
+        @resource_attribute = resource_attribute.to_s
+        @resource_class = resource_class
+        @serializer = serializer
+        @type = type.new
+      end
+
+      def block_value_for(resource:)
+        block.call(resource)
+      end
+
+      # Make nested/dot calls (e.g. `vendor.ledger_id`)
+      def dot_value_for(resource:)
+        resource_attribute_dot_parts.inject(resource) { |r, dot_method| r&.send(dot_method) }
+      end
+
+      def ledger_attribute_dot_parts
+        @ledger_attribute_dot_parts ||= ledger_attribute.split('.')
+      end
+
+      def ledger_attribute_hash_for(resource:)
+        ledger_attribute_hash_with(value: value_from_local(resource: resource))
+      end
+
+      # Create nested hash for ledger
+      #
+      # when ledger_attribute = 'Foo.Bar.Baz',
+      # ledger_attribute_hash_with(value: 123)
+      # Result:
+      # {
+      #   'Foo' => {
+      #     'Bar' => {
+      #       'Baz' => 123
+      #     }
+      #   }
+      # }
+      def ledger_attribute_hash_with(value:)
+        ledger_attribute_dot_parts.reverse.inject(value) { |a, n| { n => a } }
```

Comment:
Perhaps traverse this backwards instead of reversing and making a copy? 
Could be a useless perf optimization that though...

Reply:
Yeah, I wanted to use `inject` instead of a more complex block with `reverse_each`.